docs/financial/orderAPI.md

```diff
@@ -351,8 +353,16 @@ if (!isset($params['invoiceID']) {
 }
 $order = civicrm_api3('Order', 'create' $params);
 try {
+  // Use the Payment Processor to attempt to take the actual payment. You may
+  // pass in other params here, too.
   civicrm_api3('PaymentProcessor', 'pay', ['contribution_id' => $order['id']]);
-  civicrm_api3('Payment', 'create', ['contribution_id' => $order['id'], 'amount' => $params['amount']]);
+
+  // Assuming the payment was taken, record it which will mark the Contribution
+  // as Completed and update related entities.
+  civicrm_api3('Payment', 'create', [
+    'contribution_id' => $order['id'],
+    'total_amount' => $params['amount'],
```

Comment:
I've realised we could probably deprecate not passing paymen_processor_id or payment_instrument_id in here - perhaps we should add that to the example

Reply:
OR not AND  - we can figure out the payment_instrument_id if we have the payment_processor_id

Reply:
but yes - I think we can't really add payments without knowing what sort of payments they are without too much magic & payment method should be known when creating a payment

Reply:
Well processor_id is only relevant for processor payments -not checks. If you pass in payment_instrument_id & it is not the same as the one associated with they payment processor I think it should be ignored - meaning a  look up is still needed